api/src/main/java/org/openmrs/module/appointments/notification/impl/DefaultSmsSender.java

```diff
@@ -0,0 +1,180 @@
+package org.openmrs.module.appointments.notification.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.module.appointments.notification.SmsSender;
+import org.openmrs.util.OpenmrsUtil;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Properties;
+
+public class DefaultSmsSender implements SmsSender {
+
+    private Log log = LogFactory.getLog(this.getClass());
+
+    private static final String SMS_PROPERTIES_FILENAME = "sms-config.properties";
+    private static Properties smsSessionProperties = null;
+
+    SimpleDateFormat fmtTime = new SimpleDateFormat("HH:mm a");
+    SimpleDateFormat fmtDateTele = new SimpleDateFormat("dd/MM/yyyy");
+
+    private AdministrationService administrationService;
+
+    public DefaultSmsSender(AdministrationService administrationService) {
+        this.administrationService = administrationService;
+    }
+
+    @Override
+    public void send(String recipientMobileNumber, String patientName, String smsType, String serviceName, Date startDateTime, String teleLink) {
+        HttpURLConnection uc = null;
+        try {
+        	getSession();
+
+        	String password = "";
+            String addOnParams = "";
+            String entityId = "";
+            String templateId = "";
+            String msgText = "";
+
+            recipientMobileNumber = recipientMobileNumber.replace("+", "");
+            int recipientPhoneLengthRequired = Integer.parseInt(smsSessionProperties.getProperty("sms.phonenumber.length"));
+            if (recipientMobileNumber.length()>recipientPhoneLengthRequired) {
+            	int indexStart = recipientMobileNumber.length() - recipientPhoneLengthRequired;
+            	recipientMobileNumber = recipientMobileNumber.substring(indexStart);
+            }
+            String dateSuffixText = getDateWithSuffix(startDateTime.getDate());
+            String dateApptPattern = String.format("MMMM '%s', yyyy", dateSuffixText);
+            SimpleDateFormat fmtDateAppt = new SimpleDateFormat(dateApptPattern);
+
+            String requestUrl = smsSessionProperties.getProperty("sms.url");
```

Comment:
A lot of logic for string concatenation. Needs tests. See the previous comment on how this could possibly be restructured into an AppointmentMessage class, which constructs the message string.

Reply:
+1